### PR TITLE
Product list on category page

### PR DESCRIPTION
**Pull request by CodeNinja500**
Sorting added to category-products page working on FormControl and valueChanges.

### CODE REVIEW

File: src/app/components/category-products/category-products.component.ts [hunk 1]
```diff
@@ -7,5 +18,122 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
...
+      sort: params['sort'] ?? 'featureValue',
+      order: params['order'] ?? 'desc'
+    })),
+    shareReplay(1),
+    tap((data) => this.setControlFromQueryParams(data))
+  );
+
+  readonly sortingOpts$: Observable<SortOptionModel[]> = of([
+    { display: 'Featured', key: 'featureValue;desc', order: 'desc' },
+    { display: 'Price Low to High', key: 'price;asc', order: 'asc' },
```
Comment (gradzio) [inline src/app/components/category-products/category-products.component.ts]: if you have key, then you do not need order, or keep only the property in key

File: src/app/components/category-products/category-products.component.ts [hunk 2]
```diff
@@ -7,5 +18,122 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
...
+  );
+  readonly sortForm: FormControl = new FormControl();
+
+  readonly productsFilteredAndSorted$: Observable<ProductQueryModel[]> = combineLatest([
+    this.queryParams$,
+    this.productInCategoryList$
+  ]).pipe(
+    map(([params, products]) =>
+      products.sort((a, b) => {
+        if (params.sort === 'featureValue') {
```
Comment (gradzio) [inline src/app/components/category-products/category-products.component.ts]: too complex, you can do it without any if statement

File: src/app/components/category-products/category-products.component.ts [hunk 3]
```diff
@@ -7,5 +18,122 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
...
+      name: product.name,
+      price: product.price,
+      ratingCount: product.ratingCount,
+      ratingValue: product.ratingValue,
+      storeIds: product.storeIds,
+      ratingStars: this.mapRatingNumberToStars(product.ratingValue)
+    };
+  }
+
+  mapRatingNumberToStars(rating: number): number[] {
```
Comment (gradzio) [inline src/app/components/category-products/category-products.component.ts:101]: :+1:

File: src/app/components/category-products/category-products.component.ts (hunk 2 above)
Comment (CodeNinja500) [reply src/app/components/category-products/category-products.component.ts]: I managed to shorten this function to 2 ifs...
is that better?
`products.sort((a, b) => {
        const prev = Object.assign(a);
        const next = Object.assign(b);
        if (prev[params['sort']] > next[params['sort']]) {
          return params['order'] === 'asc' ? 1 : -1;
        }
        if (prev[params['sort']] < next[params['sort']]) {
          return params['order'] === 'asc' ? -1 : 1;
        } else return 0;
      })`

File: src/app/components/category-products/category-products.component.ts (hunk 1 above)
Comment (CodeNinja500) [reply src/app/components/category-products/category-products.component.ts]: property order deleted

**Review by gradzio — APPROVED** [review]

**Pull request merged**